Source/WebCore/rendering/style/PositionArea.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2024 Igalia S.L.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+namespace WebCore {
+
+enum CSSValueID : uint16_t;
+
+class CSSValue;
+
+/*
+(span-)left        -> horizontal, start, span?
+(span-)right       -> horizontal, end, span?
+(span-)top         -> vertical, start, span?
+(span-)bottom      -> vertical, end, span?
+
+(span-)x(self-)-start, -end     -> X, start/end, span?, self?
+(span-)y(self-)-start, -end     -> Y, start/end, span?, self?
+
+(span-)(self-)block-start, -end -> block, start/end, span?, self?
+(span-)(self-)inline-start, -end -> inline, start/end, span?, self?
+
+(span-)(self-)start, -end -> (block or inline depending on position in pair), start/end, span?, self?
+
+center      -> (opposite position of other in pair. If other is ambiguous, block), center
+span-all    -> (opposite position of other in pair. If other is ambiguous, block), span-all (or center + span ??)
+*/
+
+// 6 values, 3 bits
+enum class PositionAreaAxis {
```

Comment:
`: uint8_t`

You're not putting this in a bitfield anywhere, so the comment about bits is not applicable.

Reply:
Yeah this is when I though I'd be able to compress everything down to one byte; turns out the minimum size is 4 bytes anyway. Will remove.